app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

```diff
@@ -23,37 +23,93 @@ import androidx.navigation.findNavController
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
 import com.google.samples.apps.sunflower.PlantListFragment
-import com.google.samples.apps.sunflower.PlantListFragmentDirections
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.HomeViewPagerFragmentDirections
 import com.google.samples.apps.sunflower.data.Plant
 import com.google.samples.apps.sunflower.databinding.ListItemPlantBinding
 
 /**
  * Adapter for the [RecyclerView] in [PlantListFragment].
  */
-class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallback()) {
+class PlantAdapter : ListAdapter<Plant, RecyclerView.ViewHolder>(PlantDiffCallback()) {
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val plant = getItem(position)
-        holder.apply {
-            bind(createOnClickListener(plant.plantId), plant)
-            itemView.tag = plant
+    /**
+     * Int array of all the positions with header-style cards
+     */
+    private val headerPositions = intArrayOf(0)
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+
+        // if current position is a header element
+        if (position in headerPositions) {
+
+            // cast generic RecyclerView.ViewHolder to header view holder
+            val holderViewParams = (holder as HeaderViewHolder).itemView.layoutParams
+            val layoutParams = holderViewParams as StaggeredGridLayoutManager.LayoutParams
+
+            // set width of header to span entire screen
```

Comment:
This is a little subjective, because the documentation about method `setFullSpan` have the following description:

```When set to true, the item will layout using all span area. That means, if orientation is vertical, the view will have full width; if orientation is horizontal, the view will have full height.```